### PR TITLE
feat(longitudinal controller): add params explanation

### PR DESCRIPTION
**Pull request by takayuki5168**
Signed-off-by: Takayuki Murooka <takayuki5168@gmail.com>

## Related Issue(required)

https://github.com/autowarefoundation/autoware.universe/issues/380

## Description(required)

Added the explanation of parameters of ongitudinal controller

## Pre-Review Checklist for the PR Author

**PR Author should check the checkboxes below when creating the PR.**

- [ ] Read [commit-guidelines][commit-guidelines]
- [ ] Assign PR to reviewer

If you are adding new package following items are required:

- [ ] Documentation with description of the package is available
- [ ] A sample launch file and parameter file are available if the package contains executable nodes

## Checklist for the PR Reviewer

**Reviewers should check the checkboxes below before approval.**

- [ ] Commits are properly organized and messages are according to the guideline
- [ ] PR title describes the changes

## Post-Review Checklist for the PR Author

**PR Author should check the checkboxes below before merging.**

- [ ] All open points are addressed and tracked via issues or tickets

## CI Checks

- **Build and test for PR / build-and-test-pr**: Required to pass before the merge.
- **Build and test for PR / clang-tidy-pr**: NOT required to pass before the merge. It is up to the reviewer(s). Found false positives? See the [guidelines][clang-tidy-guidelines].
- **Check spelling**: NOT required to pass before the merge. It is up to the reviewer(s). See [here][spell-check-dict] if you want to add some words to the spell check dictionary.

[commit-guidelines]: https://www.conventionalcommits.org/en/v1.0.0/
[spell-check-dict]: https://github.com/tier4/autoware-spell-check-dict#how-to-contribute

### CODE REVIEW

File: control/trajectory_follower_nodes/design/longitudinal_controller-design.md [hunk 1]
```diff
@@ -142,10 +141,102 @@ Depending on the actuating principle of the vehicle, the mechanism that physical
...
+AutonomouStuff Lexus RX 450h for under 40 km/h driving.
+
+| Name                                 | Type   | Description                                                                                                       | Default value |
+| :----------------------------------- | :----- | :---------------------------------------------------------------------------------------------------------------- | :------------ |
+| control_rate                         | double | control rate [Hz]                                                                                                 | 30            |
+| delay_compensation_time              | double | delay for longitudinal control [s]                                                                                | 0.17          |
+| enable_smooth_stop                   | bool   | flag to enable stopping mode                                                                                      | true          |
+| enable_overshoot_emergency           | bool   | flag to enable transition to EMERGENCY when the ego is a certain distance ahead of the goal                       | true          |
```
Comment (tkimura4) [inline control/trajectory_follower_nodes/design/longitudinal_controller-design.md:154]: ahead of the goal -> ahead of the stop point ?

File: control/trajectory_follower_nodes/design/longitudinal_controller-design.md [hunk 2]
```diff
@@ -142,10 +141,102 @@ Depending on the actuating principle of the vehicle, the mechanism that physical
...
+| drive_state_offset_stop_dist        | double | The state will transit to DRIVE when the distance to the goal is larger than `drive_state_stop_dist` + `drive_state_offset_stop_dist` [m]                     | 1.0           |
+| stopping_state_stop_dist            | double | The state will transit to STOPPING when the distance to the goal is smaller than `stopping_state_stop_dist` [m]                                               | 0.5           |
+| stopped_state_entry_vel             | double | ego velocity to be considered as STOPPED state [m/s]                                                                                                          | 0.01          |
+| stopped_state_entry_acc             | double | ego acceleration to be considered as STOPPED state [m/s^2]                                                                                                    | 0.1           |
+| emergency_state_overshoot_stop_dist | double | If `enable_overshoot_emergency` is tru and the ego is `emergency_state_overshoot_stop_dist`-meter ahead of the goal, the state will transit to EMERGENCY. [m] | 1.5           |
```
Comment (tkimura4) [inline control/trajectory_follower_nodes/design/longitudinal_controller-design.md:175]: typo: tru -> true

File: control/trajectory_follower_nodes/design/longitudinal_controller-design.md [hunk 3]
```diff
@@ -142,10 +141,102 @@ Depending on the actuating principle of the vehicle, the mechanism that physical
...
+### DRIVE Parameter
+
+| Name                                  | Type   | Description                                                                                                                   | Default value |
+| :------------------------------------ | :----- | :---------------------------------------------------------------------------------------------------------------------------- | :------------ |
+| kp                                    | double | p gain for longitudinal control                                                                                               | 1.0           |
+| ki                                    | double | i gain for longidutinal control                                                                                               | 0.1           |
+| kd                                    | double | d gain for longidutinal control                                                                                               | 0.0           |
+| max_out                               | double | max value of output acceleration during DRIVE state [m/s^2]                                                                   | 1.0           |
```
Comment (tkimura4) [inline control/trajectory_follower_nodes/design/longitudinal_controller-design.md:186]: I think `max_out` is the max correction value for the original acceleration by PID.

File: control/trajectory_follower_nodes/design/longitudinal_controller-design.md [hunk 4]
```diff
@@ -142,10 +141,102 @@ Depending on the actuating principle of the vehicle, the mechanism that physical
...
+| max_i_effort                          | double | max value of acceleration with i gain                                                                                         | 0.3           |
+| min_i_effort                          | double | min value of acceleration with i gain                                                                                         | -0.3          |
+| max_d_effort                          | double | max value of acceleration with d gain                                                                                         | 0.0           |
+| min_d_effort                          | double | min value of acceleration with d gain                                                                                         | 0.0           |
+| lpf_vel_error_gain                    | double | gain of low-pass filter for velocity error                                                                                    | 0.9           |
+| current_vel_threshold_pid_integration | double | Velocity error will be integrated if absolute value of current velocity is larger larger than this parameter. [m/s]           | 0.5           |
```
Comment (tkimura4) [inline control/trajectory_follower_nodes/design/longitudinal_controller-design.md:195]: typo: larger larger  -> larger

File: control/trajectory_follower_nodes/design/longitudinal_controller-design.md [hunk 5]
```diff
@@ -142,10 +141,102 @@ Depending on the actuating principle of the vehicle, the mechanism that physical
...
+| emergency_state_traj_rot_dev        | double | If the ego's orientation differs `emergency_state_traj_rot_dev` rad from the nearest trajectory point, the state will transit to EMERGENCY. [rad]             | 0.784         |
+
+### DRIVE Parameter
+
+| Name                                  | Type   | Description                                                                                                                   | Default value |
+| :------------------------------------ | :----- | :---------------------------------------------------------------------------------------------------------------------------- | :------------ |
+| kp                                    | double | p gain for longitudinal control                                                                                               | 1.0           |
+| ki                                    | double | i gain for longidutinal control                                                                                               | 0.1           |
```
Comment (tkimura4) [inline control/trajectory_follower_nodes/design/longitudinal_controller-design.md:184]: typo:
longidutinal -> longitudinal

Comment (tkimura4) [reply control/trajectory_follower_nodes/design/longitudinal_controller-design.md:184]: & L185

File: control/trajectory_follower_nodes/design/longitudinal_controller-design.md [hunk 6]
```diff
@@ -142,10 +141,102 @@ Depending on the actuating principle of the vehicle, the mechanism that physical
...
+If the ego does not stop in a certain time or some-meter ahead of the goal, weak acceleration to stop right (`weak_stop_acc`) now will be output.
+If the ego is still running, strong acceleration (`strong_stop_acc`) to stop right now will be output.
+
+| Name                         | Type   | Description                                                                                                          | Default value |
+| :--------------------------- | :----- | :------------------------------------------------------------------------------------------------------------------- | :------------ |
+| smooth_stop_max_strong_acc   | double | max strong acceleration [m/s^2]                                                                                      | -0.5          |
+| smooth_stop_min_strong_acc   | double | min strong acceleration [m/s^2]                                                                                      | -0.8          |
+| smooth_stop_weak_acc         | double | weak accelerion [m/s^2]                                                                                              | -0.3          |
```
Comment (tkimura4) [inline control/trajectory_follower_nodes/design/longitudinal_controller-design.md:210]: typo: accelerion -> acceleration

**Comment by takayuki5168** [comment]
@tkimura4 Thanks for the comment! Fixed them in https://github.com/autowarefoundation/autoware.universe/pull/381/commits/cec511a366f995e886bbadea2b882b0ec5fd3bc8 :pray:

**Review by tkimura4 — APPROVED** [review]
LGTM
@TakaHoribe 
Please check.

File: control/trajectory_follower_nodes/design/longitudinal_controller-design.md [hunk 7]
```diff
@@ -142,10 +141,102 @@ Depending on the actuating principle of the vehicle, the mechanism that physical
...
+AutonomouStuff Lexus RX 450h for under 40 km/h driving.
+
+| Name                                 | Type   | Description                                                                                                       | Default value |
+| :----------------------------------- | :----- | :---------------------------------------------------------------------------------------------------------------- | :------------ |
+| control_rate                         | double | control rate [Hz]                                                                                                 | 30            |
+| delay_compensation_time              | double | delay for longitudinal control [s]                                                                                | 0.17          |
+| enable_smooth_stop                   | bool   | flag to enable stopping mode                                                                                      | true          |
+| enable_overshoot_emergency           | bool   | flag to enable transition to EMERGENCY when the ego is a certain distance ahead of the stop point                 | true          |
```
Comment (TakaHoribe) [inline control/trajectory_follower_nodes/design/longitudinal_controller-design.md:154]: ahead?

**Review by TakaHoribe — APPROVED** [review]

**Comment by TakaHoribe** [comment]
@takayuki5168 Thank you!

**Pull request merged**